### PR TITLE
Update CNKI translator

### PR DESCRIPTION
**Pull request by l0o0**
1. Update target regex to match CNKI url prefix via proxy
2. Add snapshot attachment by default. PDF and CAJ file links will be added when the url is accessible
3. Use `doGet` to get refoworks from CNKI. `doPost` may have CSRF
    problem via proxy.
4. Add function 'getIDFromRef' to get `dbname` and `filename` from
    reference links. `getIDFromUrl` can not get `dbname` and `filename` when
    the article is pre-released.
5. Change `journal article` field value `callNumber`. `CN` in CNKI
    refwork format is not the `callNumber`, it's a Chinese version of ISSN.
    Move `CN` to extra field.

### CODE REVIEW

**Comment by dstillman** [comment]
> Update target regex to match CNKI url prefix via proxy

[As I explained](https://forums.zotero.org/discussion/comment/341200/#Comment_341200) in the forums, this is incorrect and needs to be undone. The target is automatically compared against the unproxied URL.

File: CNKI.js [hunk 1]
```diff
@@ -41,33 +41,33 @@
...
-					//Zotero doesn't do well with mixed line endings. Make everything \n
-					.replace(/\r\n?/g, '\n')
-					//split authors
-					.replace(/^(A[1-4]|U2)\s*([^\r\n]+)/gm, function(m, tag, authors) {
+	ZU.doGet('https://kns.cnki.net/kns/ViewPage/viewsave.aspx?displayMode=Refworks' + '&' +  postData, 
+		function(text) {
+			var parser = new DOMParser();
+			var html = parser.parseFromString(text, "text/html")
+			var text = ZU.xpath(html, "//table[@class='mainTable']//td")[0].innerHTML;
+			var text = text.replace(/<br>/g, '\n');
```
Comment (dstillman) [inline CNKI.js]: Can you just use `.textContent` instead of `.innerHTML` and manual tag parsing? Replacing `<br>` manually is going to be much more fragile.

**Comment by l0o0** [comment]
https://kns.cnki.net/KCMS/detail/11.1809.S.20191010.1139.002.html, this is a pre-released article for the test.

File: CNKI.js (hunk 1 above)
Comment (l0o0) [reply CNKI.js]: I've tried to use `.textContent`, but the returned result is 'RT DissertationSR 1A1 戈亚妮A3 黄毅勤PB 首都经济贸易大学T1 中小企业融资方式的比较与选择CL 硕士YR 2018K1 中小企业;融资方式;比较分析;选择建议...`. There are no line breaks and hard to parse.  

As you mentioned in the forum, there are no proxy entries in the Proxies tab after I use the proxy.

Comment (dstillman) [reply CNKI.js]: OK, then you may need to configure the proxy manually. But the regex should just reflect the target site, without taking any proxy into account.

Comment (l0o0) [reply CNKI.js]: Could you show me some proxy config examples? I really don't know how to set proxy manually in Zotero connector.

Comment (dstillman) [reply CNKI.js]: Ask that in the forums.

File: CNKI.js [hunk 2]
```diff
@@ -173,6 +181,72 @@ function doWeb(doc, url) {
 	}
 }
 
+function scrape(ids, doc, url, itemInfo) {
```
Comment (zuphilip) [inline CNKI.js]: This function is defined twice! You should delete one of them.

File: CNKI.js [hunk 3]
```diff
@@ -232,14 +306,134 @@ function scrape(ids, doc, url, itemInfo) {
 			}
 			
 			i++;
+			
+			// CN 中国刊物编号，非refworks中的callNumber
+			if (newItem.callNumber){
+				newItem.extra = 'CN ' + newItem.callNumber;
+				newItem.callNumber = "";
```
Comment (zuphilip) [inline CNKI.js]: Add an (English) explanation as a comment why you moving this information. Moreover, I am not sure it is worth to save this journal identification number at all. Can you say a little bit more, why this could be useful for someone?

File: CNKI.js [hunk 4]
```diff
@@ -1,22 +1,22 @@
 {
 	"translatorID": "5c95b67b-41c5-4f55-b71a-48d5d7183063",
 	"label": "CNKI",
-	"creator": "Aurimas Vinckevicius",
-	"target": "^https?://([^/]+\\.)?cnki\\.net",
+	"creator": "l0o0 linxzh1989@gmail.com",
+	"target": "^https?://kns",
```
Comment (zuphilip) [inline CNKI.js]: No!

**Comment by zuphilip** [comment]
This is superseeded by #2025. However, I want to second that not continuing here but opening another PR is not very helpful for us. :angry:
